client/my-sites/checkout/checkout/domain-details-form.jsx

```diff
@@ -554,6 +556,14 @@ export class DomainDetailsFormContainer extends PureComponent {
 }
 
 export default connect(
-	state => ( { contactDetails: getContactDetailsCache( state ) } ),
+	( state ) => {
+		const contactDetails = getContactDetailsCache( state );
+		// this can one day be a selector in state/selectors if we need it elsewhere
+		const isStateRequiredInAddress = includes( CHECKOUT_US_ADDRESS_FORMAT_COUNTRIES, ( contactDetails || {} ).countryCode );
```

Comment:
I'd suggest to not use `isStateRequiredInAddress` for switching between address formats because its not very future-proof. We do want to have the ability in future to add more address formats.